project-fortis-interfaces/src/components/Insights/Maps/HeatMap.js

```diff
@@ -77,8 +79,15 @@ export default class HeatMap extends React.Component {
   componentWillReceiveProps(nextProps) {
     const { placeid, defaultZoom } = this.state;
     const { targetBbox } = this.props;
-
-    if (hasChanged(this.props, nextProps) && nextProps.selectedplace.placeid && placeid !== nextProps.selectedplace.placeid) {
+    const { sharedLink } = nextProps;
+    if (hasChanged(this.props, nextProps) && sharedLink && !this.state.sharedLinkMapRepositions) {
+      const firstBbox = nextProps.bbox.slice(0,2);
+      const secondBbox = nextProps.bbox.slice(2,4);
+      this.refs.map.leafletElement.fitBounds([firstBbox, secondBbox]);
+      this.setState({sharedLinkMapRepositions: true});
+      this.setState({mapMovedByProps: false});
```

Comment:
I think it should be possible to remove the mapMovedByProps and related conditionals now. Please double check if this is possible.

Reply:
@c-w yup, still good.